### PR TITLE
Rhmap 16530 release finish pipeline

### PR DESCRIPTION
**Pull request by mikenairn**
* Update getReleaseTag helper to make candidate optional, if null (default) it isn't an rc tag.
* Add tagRepo and tagBranch helpers.

### CODE REVIEW

File: vars/branchRepo.groovy [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/groovy
+
+def call(String gitRepoUrl, String gitRef, String checkoutDir, String branchName, Closure body) {
+    checkoutGitRepo {
+        repoUrl = gitRepoUrl
+        branch = gitRef
+        targetDir = checkoutDir
+    }
+    dir(checkoutDir) {
+        sshagent(['jenkinsgithub']) {
```
Comment (grdryn) [inline vars/branchRepo.groovy:10]: Would it be worthwhile to make the credential name here, and the git user.name and email below, configurable?

**Review by grdryn — APPROVED** [review]

File: vars/branchRepo.groovy (hunk 1 above)
Comment (mikenairn) [reply vars/branchRepo.groovy:10]: Yes, was already thinking about that. I don't however want to add all these to the signature so might add a config/parameters object like we have done in others e.g https://github.com/feedhenry/fh-pipeline-library/blob/master/vars/fhBuildNode.groovy#L3.

Comment (mikenairn) [reply vars/branchRepo.groovy:10]: @grdryn I'm going to leave this for now as i want to think about the best way to do this, we have the same problem in a few places so want to come up with a consistent solution.

File: vars/branchRepo.groovy [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/groovy
+
+def call(String gitRepoUrl, String gitRef, String checkoutDir, String branchName, Closure body = {}) {
```
Comment (mikenairn) [inline vars/branchRepo.groovy:3]: @grdryn Added a default empty body object to both tagRepo and branchRepo methods to make this optional.

**Review by grdryn — APPROVED** [review]

**Pull request merged**